proxy/src/http/sql_over_http.rs

```diff
@@ -192,15 +200,51 @@ pub async fn handle(
     // Read the query and query params from the request body
     //
     let body = hyper::body::to_bytes(request.into_body()).await?;
-    let QueryData { query, params } = serde_json::from_slice(&body)?;
-    let query_params = json_to_pg_text(params)?;
+    let payload: Payload = serde_json::from_slice(&body)?;
+
+    let mut client = conn_pool.get(&conn_info, !allow_pool).await?;
 
     //
     // Now execute the query and return the result
     //
-    let client = conn_pool.get(&conn_info, !allow_pool).await?;
+    let result = match payload {
+        Payload::Single(query) => query_to_json(&client, query, raw_output, array_mode).await,
+        Payload::Batch(queries) => {
+            let mut results = Vec::new();
+            let transaction = client.transaction().await?;
+            for query in queries {
+                let result = query_to_json(&transaction, query, raw_output, array_mode).await;
+                if let Err(e) = result {
+                    transaction.rollback().await?;
+                    return Err(e.into());
+                }
+                results.push(result?);
```

Comment:
I think something like this is a bit more correct (albeit equivalent)
```suggestion
                match result {
                    Ok(r) => results.push(r),
                    Err(e) => {
                        transaction.rollback().await?;
                        return Err(e.into());
                    }
                }
```

Reply:
fixed in https://github.com/neondatabase/neon/pull/4654/commits/4346b1445740619131c43521e740fa9ad59c1ff0